kit/src/pool/geometric_mean.rs

```diff
@@ -1 +1,131 @@
+use bindings::{geometric_mean::GeometricMean, geometric_mean_solver::GeometricMeanSolver};
+use ethers::types::Address;
 
+use super::*;
+
+pub struct GeometricMeanPool {
+    pub strategy_contract: GeometricMean<ArbiterMiddleware>,
+    pub solver_contract: GeometricMeanSolver<ArbiterMiddleware>,
+    pub parameters: GeometricMeanParameters,
+}
+
+pub struct GeometricMeanParameters {
+    pub swap_fee: eU256,
+    pub update_parameters: UpdateParameters,
+}
+
+pub enum UpdateParameters {
+    SwapFee(eU256),
+    TargetTimestamp(eU256),
+    TargetWeightX(eU256),
+    TargetController(Address),
+}
+
+pub enum GeometricMeanAllocationData {
+    AllocateGivenX(eU256),
+    AllocateGivenY(eU256),
+    DeallocateGivenX(eU256),
+    DeallocateGivenY(eU256),
+}
+
+impl PoolType for GeometricMeanPool {
+    type UpdateParameters = GeometricMeanParameters;
+    type StrategyContract = GeometricMean<ArbiterMiddleware>;
+    type SolverContract = GeometricMeanSolver<ArbiterMiddleware>;
+    type AllocationData = GeometricMeanAllocationData;
+
+    async fn swap_data(
+        &self,
+        pool_id: eU256,
+        input_token: InputToken,
+        amount_in: eU256,
+    ) -> Result<Bytes> {
+        let (valid, _, _, data) = match input_token {
+            InputToken::TokenX => {
+                self.solver_contract
+                    .simulate_swap(pool_id, true, amount_in)
+                    .call()
+                    .await?
+            }
+            InputToken::TokenY => {
+                self.solver_contract
+                    .simulate_swap(pool_id, false, amount_in)
+                    .call()
+                    .await?
+            }
+        };
+        if valid {
+            Ok(data)
+        } else {
+            anyhow::bail!("swap was invalid!")
+        }
+    }
+
+    async fn update_data(&self, new_data: Self::UpdateParameters) -> Result<Bytes> {
+        let data = match new_data.update_parameters {
+            UpdateParameters::SwapFee(fee) => {
+                self.solver_contract.prepare_fee_update(fee).call().await?
+            }
+            UpdateParameters::TargetTimestamp(timestamp) => {
+                self.solver_contract
+                    .prepare_weight_x_update(timestamp, timestamp)
+                    .call()
+                    .await?
+            }
+            UpdateParameters::TargetWeightX(weight_x) => {
+                self.solver_contract
+                    .prepare_weight_x_update(weight_x, weight_x)
+                    .call()
+                    .await?
+            }
```

Comment:
The target timestamp and the target weight must be passed together to the `prepareWeightXUpdate` function, see the interface:

```solidity
function prepareWeightXUpdate(
    uint256 targetWeightX,
    uint256 targetTimestamp
) public pure returns (bytes memory);
```